alicloud/service_alicloud_gpdb.go

```diff
@@ -101,6 +101,42 @@ func (s *GpdbService) ModifyGpdbSecurityIps(id, ips string) error {
 	return nil
 }
 
+func (s *GpdbService) DescribeGpdbConnection(id string) (*gpdb.DBInstanceNetInfo, error) {
+	parts, err := ParseResourceId(id, 2)
+	if err != nil {
+		return nil, WrapError(err)
+	}
+
+	// Describe DB Instance Net Info
+	request := gpdb.CreateDescribeDBInstanceNetInfoRequest()
+	request.DBInstanceId = parts[0]
+	raw, err := s.client.WithGpdbClient(func(gpdbClient *gpdb.Client) (interface{}, error) {
+		return gpdbClient.DescribeDBInstanceNetInfo(request)
+	})
+	if err != nil {
+		if IsExceptedErrors(err, []string{InvalidGpdbInstanceIdNotFound, InvalidGpdbNameNotFound}) {
+			return nil, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+		}
+		return nil, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw)
+	response, _ := raw.(*gpdb.DescribeDBInstanceNetInfoResponse)
+	if len(response.DBInstanceNetInfos.DBInstanceNetInfo) < 1 {
+		return nil, WrapErrorf(Error(GetNotFoundMessage("GpdbConnection", id)), NotFoundMsg, ProviderERROR)
+	}
+
+	object := response.DBInstanceNetInfos.DBInstanceNetInfo
+	if object != nil {
+		for _, o := range object {
+			if strings.HasPrefix(o.ConnectionString, parts[1]) {
+				return &o, nil
+			}
+		}
+	}
+
```

Comment:
I think line 124-136 can be merged to `for _, o := range response.DBInstanceNetInfos.DBInstanceNetInfo`